internal/receiver/scriptedinputsreceiver/factory.go

```diff
@@ -0,0 +1,68 @@
+// Copyright Splunk, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scriptedinputreceiver
+
+import (
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/adapter"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/receiver"
+
+	"github.com/signalfx/splunk-otel-collector/internal/receiver/scriptedinputsreceiver/procpipe"
+)
+
+const (
+	typeStr   = "scripted_inputs"
+	stability = component.StabilityLevelDevelopment
+)
+
+func NewFactory() receiver.Factory {
+	return adapter.NewFactory(ReceiverType{}, stability)
+}
+
+// ReceiverType implements stanza.LogReceiverType
+// to create a file tailing receiver
```

Comment:
I think conventionally this is demonstrated/enforced with:

```suggestion
var _ adapter.LogReceiverType = (*ReceiverType)(nil)

```